chore: Add 4.4.0 release notes (#1180)
medic/cht-docs: content/en/core/releases/4.4.0.md

---
title: "4.4.0 release notes"
linkTitle: "4.4.0"
weight:
description: >
relevantLinks: >
toc_hide: true
---

## Known issues

Check the repository for the [latest known issues](https://github.com/medic/cht-core/issues?q=is%3Aissue+label%3A%22Affects%3A+4.4.0%22).

## Upgrade notes

### Breaking changes

None.

### UI/UX changes

#### CHT dialogs.

> CHT dialogs were updated to align with Material guidelines. This includes updating the visual design / placement of dialogs as well as updating the default text of dialogs as necessary.

Reviewer: ```suggestion
CHT dialogs were updated to align with [Material guidelines](https://m3.material.io/components/dialogs/guidelines). This includes updating the visual design and text of dialogs to align with best practices.
```

Author: Thank you! ✨

{{< figure src="../images/4.4.0-8391-cht-dialogs.png" class="left col-12 col-lg-12" >}}
<br clear="all">

- [#8390](https://github.com/medic/cht-core/issues/8390): Align "dialog" components with Material design guidelines

## Highlights

### Modals\'s text

> Some modals\'s text have been improved to increase understanding and some keys have been replaced.
> 
> - Bulk delete reports
>   - `bulkdelete.confirm.title` used for the modal title when deleting 1 report
>   - [NEW] `bulkdelete.confirm.title.plural` used for the modal title when deleting multiple reports.
> - Edit message group
>   - [NEW] `edit_message_group.modal.title`: used for the modal title
> - Report a bug
>   - [NEW] `feedback.modal.submit`: used for the modal submit button label. 
> - Confirm Navigation - like when exiting a form that hasn't been submitted
>   - [NEW] `confirm.destructive.navigation.title`: used for the modal title
>   - [NEW] `confirm.destructive.navigation.submit`: used for the modal submit button label.
> - Training cards
>   - [NEW] `training_cards.confirm.title`: used for the modal title
> - Confirm verifying a report
>   - [NEW] `confirm.verification.title`: used for the modal title
>   - [NEW] `confirm.verification.submit`: used for the modal submit button label.
> 
> We have provided translations for all of these in the CHT Core supported languages (English, Spanish, French, Indonesian, Hindi, Nepali, Bambara, Swahili)

Reviewer: ```suggestion
As part of this work, some translation keys changed. We have provided translations for all of the languages officially supported by CHT (English, Spanish, French, Indonesian, Hindi, Nepali, Bambara, Swahili). If you have a custom language and have any of these features, you will need to translate these in Admin app > Display > Translations tab. 

- Bulk delete reports dialog
  - `bulkdelete.confirm.title`: used for the modal title when deleting one report
  - [NEW] `bulkdelete.confirm.title.plural`: used for the modal title when deleting multiple reports.
- Edit SMS message group dialog
  - [NEW] `edit_message_group.modal.title`: used for the modal title
- Report a bug dialog
  - [NEW] `feedback.modal.submit`: used for the modal submit button label. 
- Navigation when form not saved dialog
  - [NEW] `confirm.destructive.navigation.title`: used for the modal title
  - [NEW] `confirm.destructive.navigation.submit`: used for the modal submit button label.
- Training cards dialog
  - [NEW] `training_cards.confirm.title`: used for the modal title
- Confirmation when verifying a report dialog
  - [NEW] `confirm.verification.title`: used for the modal title
  - [NEW] `confirm.verification.submit`: used for the modal submit button label.
```
I think better move this to line 29 because is about the same work.

Author: Done 👍

### Features

None.

### Improvements

- [#6514](https://github.com/medic/cht-core/issues/6514): Migrate from SW-Precache to Workbox
- [#8179](https://github.com/medic/cht-core/issues/8179): nginx and haproxy should respect `Accept` headers when responding with error templates
- [#8390](https://github.com/medic/cht-core/issues/8390): Align "dialog" components with Material design guidelines
- [#8442](https://github.com/medic/cht-core/issues/8442): Telemetry geolocation:failure events for geolocation should be unique
- [#8498](https://github.com/medic/cht-core/issues/8498): Remove targets disable admin option
- [#8506](https://github.com/medic/cht-core/issues/8506): Update browser support notice to match official versions
- [#8511](https://github.com/medic/cht-core/issues/8511): Update configuration deployment action to use latest cht-conf

### Security fixes

> - [#8443](https://github.com/medic/cht-core/issues/8443): Security issue 8156

Reviewer: ```suggestion
- [#8443](https://github.com/medic/cht-core/issues/8443): Masking credentials
```

### Performance improvements

> - [#8238](https://github.com/medic/cht-core/issues/8238): Figure out a different way to use purged docs cache

Reviewer: Here this can be better explained as: `store purged docs cache in couchdb `
I think this is something that app builders want to know a bit more about, from the PR: `each user gets their own doc in a new database medic-cache to store purged document ids the database is destroyed (deleted and recreated) every time purging runs`

Author: Thank you for the details! ✨

- [#8451](https://github.com/medic/cht-core/issues/8451): Add support for HTTP/2 in nginx 

### Bug fixes

- [#6180](https://github.com/medic/cht-core/issues/6180): Any text containing back ticks looks weird in the webapp

> - [#8081](https://github.com/medic/cht-core/issues/8081): `/api/v2/export/contacts` does not return useful contact type data for custom hierarchy

Reviewer: I think here: `Add contact_type and place_id to contacts exported by /api/v2/export/contacts`

> - [#8243](https://github.com/medic/cht-core/issues/8243): Can't turn off permissions for `national_admin` role in admin app

Reviewer: To fix this, we removed the `national_admin` role from our code source; maybe this should be highlighted and let the app builders know. What do you think?

Author: Yes, that is also a suggestion from Diana in [this comment](https://github.com/medic/cht-docs/pull/1180#issuecomment-1718764813)
Thank you!

### Technical improvements

- [#6132](https://github.com/medic/cht-core/issues/6132): Files containing tests should use the .spec.js extension
- [#6289](https://github.com/medic/cht-core/issues/6289): Upgrade to CouchDB v3
- [#8306](https://github.com/medic/cht-core/issues/8306): Replace grunt with npm scripts
- [#8414](https://github.com/medic/cht-core/issues/8414): Flaky test: sms-gateway api returns list and updates state
- [#8459](https://github.com/medic/cht-core/issues/8459): Update nginx to latest 
- [#8469](https://github.com/medic/cht-core/issues/8469): Scalability suite fails because Jmeter version no longer exists
- [#8494](https://github.com/medic/cht-core/issues/8494): Update lodash
- [#8516](https://github.com/medic/cht-core/issues/8516): Remove unused /api/auth/ endpoint
- [#8518](https://github.com/medic/cht-core/issues/8518): Encode parameters being used in credentials url
- [#8522](https://github.com/medic/cht-core/issues/8522): Flaky test: More Options Menu - Offline User all permissions disabled
- [#8524](https://github.com/medic/cht-core/issues/8524): Remove sub-scripts from package.json

> - [#8533](https://github.com/medic/cht-core/issues/8533): Builds failing pushing build history to git repo

Reviewer: ```suggestion
- [#8533](https://github.com/medic/cht-core/issues/8533): Builds failing pushing build history to git repo
```
I read it 3 times processing the meaning 😅 what about: `Ignore unimportant step failures in CI build`

- Bulk delete reports dialog
  - `bulkdelete.confirm.title`: used for the modal title when deleting one report
  - [NEW] `bulkdelete.confirm.title.plural`: used for the modal title when deleting multiple reports.
- Edit SMS message group dialog
  - [NEW] `edit_message_group.modal.title`: used for the modal title
- Report a bug dialog
  - [NEW] `feedback.modal.submit`: used for the modal submit button label.
- Navigation when form not saved dialog
  - [NEW] `confirm.destructive.navigation.title`: used for the modal title
  - [NEW] `confirm.destructive.navigation.submit`: used for the modal submit button label.
- Training cards dialog
  - [NEW] `training_cards.confirm.title`: used for the modal title
- Confirmation when verifying a report dialog
  - [NEW] `confirm.verification.title`: used for the modal title
  - [NEW] `confirm.verification.submit`: used for the modal submit button label.

- [#8390](https://github.com/medic/cht-core/issues/8390): Align "dialog" components with Material design guidelines

## Highlights

### Store purged docs cache in couchdb

> Each user gets their own doc in a new database `medic-cache` to store purged document ids the database is destroyed (deleted and recreated) every time purging runs

Reviewer: ```suggestion
Each user gets their own doc in a new database `medic-cache` to store purged document IDs. The database is destroyed (deleted and recreated) every time purging runs.
```

- Bulk delete reports dialog
  - `bulkdelete.confirm.title`: used for the modal title when deleting one report
  - [NEW] `bulkdelete.confirm.title.plural`: used for the modal title when deleting multiple reports.
- Edit SMS message group dialog
  - [NEW] `edit_message_group.modal.title`: used for the modal title
- Report a bug dialog
  - [NEW] `feedback.modal.submit`: used for the modal submit button label.
- Navigation when form not saved dialog
  - [NEW] `confirm.destructive.navigation.title`: used for the modal title
  - [NEW] `confirm.destructive.navigation.submit`: used for the modal submit button label.
- Training cards dialog
  - [NEW] `training_cards.confirm.title`: used for the modal title
- Confirmation when verifying a report dialog
  - [NEW] `confirm.verification.title`: used for the modal title
  - [NEW] `confirm.verification.submit`: used for the modal submit button label.

- [#8390](https://github.com/medic/cht-core/issues/8390): Align "dialog" components with Material design guidelines

## Highlights

### Store purged docs cache in couchdb

> Each user gets their own doc in a new database `medic-cache` to store purged document ids the database is destroyed (deleted and recreated) every time purging runs

Reviewer 2: Let's make it clear this is a performance improvement so people know why they should be excited! @dianabarsan Do we have any performance numbers we can show off?

Reviewer 3: I don't have numbers. I don't think I can get the numbers in a timely manner. 
All I have is a prior analysis of how much these requests cost in a live instance and the fact that we don't make them anymore.

Reviewer: > All I have is a prior analysis of how much these requests cost in a live instance

That still sounds like good info to give an idea of the improvement, you can drop it here

### Can't turn off permissions for `national_admin` role in admin app.

Removes hardcoded conditions making national_admin role sometimes behave as CouchDb _admin.

> Adds `national_admin` as role in default config for backwards compatibility.

Reviewer 2: I don't think this is a highlight. Hopefully it doesn't impact anyone, and if it does it's probably a bug.

Author: Thanks!
I will move it to the `Upgrade notes` section

- [#8511](https://github.com/medic/cht-core/issues/8511): Update configuration deployment action to use latest cht-conf

### Security fixes

- [#8443](https://github.com/medic/cht-core/issues/8443): Masking credentials

### Performance improvements

- [#8238](https://github.com/medic/cht-core/issues/8238): Figure out a different way to use purged docs cache
- [#8451](https://github.com/medic/cht-core/issues/8451): Add support for HTTP/2 in nginx 

### Bug fixes

- [#6180](https://github.com/medic/cht-core/issues/6180): Any text containing back ticks looks weird in the webapp
- [#8081](https://github.com/medic/cht-core/issues/8081): Add contact_type and place_id to contacts exported by `/api/v2/export/contacts`
- [#8243](https://github.com/medic/cht-core/issues/8243): Can't turn off permissions for `national_admin` role in admin app
- [#8490](https://github.com/medic/cht-core/issues/8490): API crashes when CouchDB is overloaded
- [#8527](https://github.com/medic/cht-core/issues/8527): Unable to build nginx with symlinks in some environments

### Technical improvements

- [#6132](https://github.com/medic/cht-core/issues/6132): Files containing tests should use the .spec.js extension

> - [#6289](https://github.com/medic/cht-core/issues/6289): Upgrade to CouchDB v3

Reviewer 2: This should be a highlight and a performance improvement (not a technical improvement)... I've updated the issue labels to reflect this.

Author: Ticket `#6289` was moved to the `Performance improvements` section.
I asked @dianabarsan's help to add a more details explanation of the ticket to add it in the `Highlights` section

---
title: "4.4.0 release notes"
linkTitle: "4.4.0"
weight:
description: >
relevantLinks: >
toc_hide: true
---

## Known issues

Check the repository for the [latest known issues](https://github.com/medic/cht-core/issues?q=is%3Aissue+label%3A%22Affects%3A+4.4.0%22).

## Upgrade notes

The deprecated hardcoded `national_admin` role no longer behaves as a CouchDb admin. The role now behaves as an ordinary online user role, with no additional implicit permissions.
Configurations using `national_admin` role can either switch to using a CouchDb admin for admin-only tasks and/or granting more permissions to `national_admin`.

### Breaking changes

None.

### UI/UX changes

#### CHT dialogs.

CHT dialogs were updated to align with [Material guidelines](https://m3.material.io/components/dialogs/guidelines). This includes updating the visual design and text of dialogs to align with best practices.

{{< figure src="../images/4.4.0-8391-cht-dialogs.png" class="left col-12 col-lg-12" >}}
<br clear="all">

> As part of this work, some translation keys changed. We have provided translations for all of the languages officially supported by CHT (English, Spanish, French, Indonesian, Hindi, Nepali, Bambara, Swahili). If you have a custom language and have any of these features, you will need to translate these in Admin app > Display > Translations tab.

Reviewer 2: I don't recommend translating via the admin app because if the app is then updated using cht-conf these changes will get overridden.

```suggestion
As part of this work, some translation keys changed. We have provided translations for all of the languages officially supported by CHT (English, Spanish, French, Indonesian, Hindi, Nepali, Bambara, Swahili). If you have a custom language and have any of these features, you will need to translate these in your configuration files.
```

- Bulk delete reports dialog
  - `bulkdelete.confirm.title`: used for the modal title when deleting one report
  - [NEW] `bulkdelete.confirm.title.plural`: used for the modal title when deleting multiple reports.
- Edit SMS message group dialog
  - [NEW] `edit_message_group.modal.title`: used for the modal title
- Report a bug dialog
  - [NEW] `feedback.modal.submit`: used for the modal submit button label.
- Navigation when form not saved dialog
  - [NEW] `confirm.destructive.navigation.title`: used for the modal title
  - [NEW] `confirm.destructive.navigation.submit`: used for the modal submit button label.
- Training cards dialog
  - [NEW] `training_cards.confirm.title`: used for the modal title
- Confirmation when verifying a report dialog
  - [NEW] `confirm.verification.title`: used for the modal title
  - [NEW] `confirm.verification.submit`: used for the modal submit button label.

- [#8390](https://github.com/medic/cht-core/issues/8390): Align "dialog" components with Material design guidelines

## Highlights

### Upgrading CouchDb to v3.3.2

> CouchDb 3 has been released in 2020. It packs security fixes, performance and scalability improvements and a plethora of bug fixes.

Reviewer 2: ```suggestion
CouchDb 3 was released in 2020. It packs security fixes, performance and scalability improvements and a plethora of bug fixes.
```
